src/cpp/shared/game/save.cpp

```diff
@@ -42,6 +42,18 @@ extern void __fastcall GenerateStandardFileName(char*, char*);
 static const char *tmpFileName = "tmp";
 static const LPCWSTR tmpFileNameW = L"tmp";
 
+template<typename T>
+void ReadArrayFromXML(T &dest, xsd::cxx::tree::sequence<ironfist_save::arrayInt_t> &src) {
+  for (int i = 0; i < src.size(); i++)
+    dest[i] = src.at(i).value();
+}
+
+template<typename T>
+void WriteArrayToXML(xsd::cxx::tree::sequence<ironfist_save::arrayInt_t> &dest, T &src) {
+  for (int i = 0; i < ELEMENTS_IN(src); i++)
```

Comment:
#include<stdio.h>

#define ELEMENTS_IN(x) ((sizeof(x))/(sizeof((x)[0])))


int foo[42];

template<typename T>
void doThing(T &src) {
  printf("Size is %d\n", ELEMENTS_IN(src));
}


int main() {
  doThing(foo);
}


-----------------

Output: 42

Mind blown. I am so confused.